k8s/client.go

```diff
@@ -53,7 +55,29 @@ type client struct {
 }
 
 // New creates a client for the kubernetes apiserver.
-func New(apiServerURL string, caCert []byte, token string) (Client, error) {
+func New(caCertFile string, tokenFile string, apiServer string) Client {
+	caCert := readFile(caCertFile)
+	token := string(readFile(tokenFile))
```

Comment:
I don't agree moving file reading inside of the k8s library. It makes it less generic and harder to test. The files are more about how the binary itself interacts with an external system. We could just as easily change it to take it as a  string of the token rather than a filename, and I don't think that should require changing the k8s library. Also consider if another project depends on the k8s library, we shouldn't force them to use a file.